diesel/src/serialize.rs

```diff
@@ -30,43 +30,34 @@ pub enum IsNull {
 }
 
 /// Wraps a buffer to be written by `ToSql` with additional backend specific
-/// utilities.
-#[derive(Clone, Copy)]
+/// utilities.y
 pub struct Output<'a, T, DB>
 where
     DB: TypeMetadata,
     DB::MetadataLookup: 'a,
 {
     out: T,
-    metadata_lookup: Option<&'a DB::MetadataLookup>,
+    metadata_lookup: Option<&'a mut DB::MetadataLookup>,
```

Comment:
It looks like this should always be there so it shouldn't be an `Option`.
```suggestion
    metadata_lookup: &'a mut DB::MetadataLookup,
```

(and
```rust
    /// Construct a new `Output`
    pub fn new(out: T, metadata_lookup: &'a mut DB::MetadataLookup) -> Self {
        Output {
            out,
            metadata_lookup,
        }
    }

    /// Create a new `Output` with the given buffer
    pub fn with_buffer<U>(&mut self, new_out: U) -> Output<U, DB> {
        Output {
            out: new_out,
            metadata_lookup: &mut *self.metadata_lookup,
        }
    }

    /// Returns the backend's mechanism for dynamically looking up type
    /// metadata at runtime, if relevant for the given backend.
    pub fn metadata_lookup(&mut self) -> &mut DB::MetadataLookup {
        &mut *self.metadata_lookup
    }
```
)

Reply:
It's always be there, but not for [some test code](https://github.com/diesel-rs/diesel/blob/c0b613024adf970dc888453c984d42e86bad6b3e/diesel/src/serialize.rs#L73-L83).  As those tests would be much harder to write if they would require an actual connection we've choosen to just go for the option back then.